src/core/audio/AudioFile.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * AudioFile.cpp
+ *
+ * Copyright (c) 2025 Sotonye Atemie <sakertooth@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "AudioFile.h"
+
+#include <sndfile.h>
+
+namespace {
+using namespace lmms;
+
+SNDFILE* openAudioFile(std::filesystem::path path, int mode, SF_INFO* sfinfo)
```

Comment:
```suggestion
SNDFILE* openAudioFile(const std::filesystem::path& path, int mode, SF_INFO* sfinfo)
```

Reply:
It owns the underlying string, so it isn't cheap to copy. There's a proposal to add a lightweight `std::filesystem::path_view` but it probably won't be available until C++29.